### PR TITLE
xds interop: Update Docker images to `eclipse-temurin:11-jre`

### PR DESCRIPTION
**Pull request by sergiitk**
- Update to xDS Test Client and xDS test server Docker images to `eclipse-temurin:11-jre`. 
- Perform software update so that we install patches for latest vulnerabilities.

### CODE REVIEW

**Comment by sergiitk** [comment]
Test run: 
- [ ] [grpc/java/master/branch/psm-security](https://source.cloud.google.com/results/invocations/516cc571-79b9-4909-a700-2c197cc05284)

**Comment by sergiitk** [comment]
Runs above failed for unrelated reason. Now trying `eclipse-temurin:11.0.19_7-jre` per @ejona86's recommendation.

**Comment by sergiitk** [comment]
`eclipse-temurin:11.0.19_7-jre`, added Ubuntu packages update:

- [ ] [grpc/java/master/branch/psm-security](https://source.cloud.google.com/results/invocations/2391e913-8daf-4b7f-a777-d464bbf71c02)

**Comment by sergiitk** [comment]
Doing apt-get update resolved the freetype [CVE-2023-2004](http://people.ubuntu.com/~ubuntu-security/cve/CVE-2023-2004).

File: buildscripts/xds-k8s/test-client.Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Build runtime image.
-FROM openjdk:11.0.9.1-jdk
+FROM eclipse-temurin:11.0.19_7-jre
```
Comment (ejona86) [inline buildscripts/xds-k8s/test-client.Dockerfile:2]: Even with the apt-get update, we're going to need to regularly update the version here on all branches. What is the hoped for process for that?

File: buildscripts/xds-k8s/test-client.Dockerfile [hunk 2]
```diff
@@ -11,5 +11,12 @@ COPY grpc-interop-testing/ $APP_DIR/
 COPY logging*.properties $APP_DIR/
 ENV JAVA_OPTS="-Djava.util.logging.config.file=$APP_DIR/logging-json.properties"
 
+# Intentially after the app COPY to force the update on each build.
```
Comment (ejona86) [inline buildscripts/xds-k8s/test-client.Dockerfile:14]: s/Intentially/Intentionally/

File: buildscripts/xds-k8s/test-client.Dockerfile (hunk 1 above)
Comment (sergiitk) [reply buildscripts/xds-k8s/test-client.Dockerfile:2]: Yes, makes sense. You're right.

**Comment by sergiitk** [comment]
New runs, `11-jre`:
- [ ] [grpc/java/master/branch/psm-security](https://source.cloud.google.com/results/invocations/)
- [ ] TBD

**Review by ejona86 — APPROVED** [review]

**Pull request merged**